### DISCUSSION TITLE
mode=0 means

### DISCUSSION THREAD

**Discussion by BoyuanTang331** [InteractiveComputerGraphics/SPlisHSPlasH#288] [Q&A] [answered]
hi,

`scene.fluidBlocks.append(Scenes.FluidBlock(id='Fluid', boxMin = [-1.5, 0.0, -1.5], boxMax = [-0.5, 2.0, -0.5], mode=0, initialVelocity=[0.0, 0.0, 0.0]))`

in this code the initial condition of fluid is defined in fluidBlocks, what is the mode here means, 

Best,

**Comment by BoyuanTang331** [comment]
is that represent for the sampling rate, how will this behaviour influence the simulation?

denseMode (int):
0: regular sampling
1: more dense sampling
2: dense sampling

**Reply by janbender** [reply, marked as answer]
Yes, it just defines how the fluid block is sampled. If you have a sparser sampling, then at the beginning of the simulation the block will be compressed a bit until you reach the rest density. So that is the only difference.
